### PR TITLE
Fix: don't document private macros by default

### LINKED ISSUE #88453: Private macros are documented
### Code

I tried this code:

```rust
macro_rules! private_macro {
    () => {};
}
```

I expected to see this happen: `private_macro` is hidden from the documentation.

Instead, this happened: `private_macro` is shown in the documentation and has its own page. Private declarative macros are also shown.

### Version it worked on

It most recently worked on: rustc 1.56.0-nightly (8007b506a 2021-08-14)

### Version with regression

`rustc --version`:
```
rustc 1.56.0-nightly (5eacec9ec 2021-08-28)
```

### PR DESCRIPTION
**Pull request by inquisitivecrystal**
As part of #88019, I made it so private macros are documented in `--document-private-items` mode. Unfortunately, it appears that I also accidentally made them be documented when *not* in `--document-private-items` mode. This PR fixes that and adds a regression test.

r? @jyn514 (I hope you don't mind that I keep sending PRs your way)

Fixes #88453.

### CODE REVIEW

File: src/librustdoc/passes/stripper.rs [hunk 1]
```diff
@@ -70,8 +71,8 @@ impl<'a> DocFolder for Stripper<'a> {
 
             clean::ImplItem(..) => {}
 
-            // tymethods/macros have no control over privacy
-            clean::MacroItem(..) | clean::TyMethodItem(..) => {}
+            // tymethods have no control over privacy
+            clean::TyMethodItem(..) => {}
 
             // Proc-macros are always public
```
Comment (jyn514) [inline src/librustdoc/passes/stripper.rs:77]: Should this also be changed for proc macros? Can you add a test? You can add a dependency using // aux-build in a UI test.

File: src/test/rustdoc/macro-private-not-documented.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Checks that private macros aren't documented by default. They
+// should be still be documented in `--document-private-items` mode,
+// but that's tested in `macro-document-private.rs`.
```
Comment (jyn514) [inline src/test/rustdoc/macro-private-not-documented.rs:3]: nit: Can you combine this file with the test you wrote in the other PR? Rustdoc tests are really slow to run and I haven't gotten around to investigating it yet.

Comment (inquisitivecrystal) [reply src/test/rustdoc/macro-private-not-documented.rs:3]: Which one? This fits best logically with `macro-document-private.rs`, but that needs `--document-private-items` and this one needs not to have that option. The only rustdoc test I added that doesn't have that option is `macro-indirect-use.rs`, but this doesn't fit very well there.

File: src/librustdoc/passes/stripper.rs (hunk 1 above)
Comment (inquisitivecrystal) [reply src/librustdoc/passes/stripper.rs:77]: Probably not? Proc macros are always public, as far as I can tell. An import of a proc macro could still be private, but I'm not sure if that needs to be handled separately.

File: src/test/rustdoc/macro-private-not-documented.rs (hunk 2 above)
Comment (jyn514) [reply src/test/rustdoc/macro-private-not-documented.rs:3]: Ah ok, fair enough. No need to worry about it.

**Pull request merged**